packages/cdktf/lib/terraform-provisioner.ts

```diff
@@ -0,0 +1,380 @@
+import { ref } from "./tfExpression";
+import { Token } from "./tokens/token";
+import { snakeCase } from "./util";
+/**
+ * Most provisioners require access to the remote resource via SSH or WinRM and expect a nested connection block with details about how to connect.
+ *
+ * See {@link https://www.terraform.io/language/resources/provisioners/connection connection}
+ */
+export interface ISSHProvisionerConnection {
+  /**
+   * The connection type. Valid values are "ssh" and "winrm".
+   * Provisioners typically assume that the remote system runs Microsoft Windows when using WinRM.
+   * Behaviors based on the SSH target_platform will force Windows-specific behavior for WinRM, unless otherwise specified.
+   */
+  readonly type: "ssh";
+
+  /**
+   * The user to use for the connection.
+   *
+   * @default root
+   */
+  readonly user?: string;
+
+  /**
+   * The password to use for the connection.
+   */
+  readonly password?: string;
+
+  /**
+   * The address of the resource to connect to.
+   */
+  readonly host: string;
+
+  /**
+   * The port to connect to.
+   *
+   * @default 22
+   */
+  readonly port?: number;
+
+  /**
+   * The timeout to wait for the connection to become available.
+   * Should be provided as a string (e.g., "30s" or "5m".)
+   *
+   * @default 5m
+   */
+  readonly timeout?: string;
+
+  /**
+   * The path used to copy scripts meant for remote execution.
+   * Refer to {@link https://www.terraform.io/language/resources/provisioners/connection#how-provisioners-execute-remote-scripts How Provisioners Execute Remote Scripts below for more details}
+   */
+  readonly scriptPath?: string;
+
+  /**
+   * The contents of an SSH key to use for the connection.
+   * These can be loaded from a file on disk using the file function.
+   * This takes preference over password if provided.
+   */
+  readonly privateKey?: string;
+
+  /**
+   * The contents of a signed CA Certificate.
+   * The certificate argument must be used in conjunction with a private_key.
+   * These can be loaded from a file on disk using the the file function.
+   */
+  readonly certificate?: string;
+
+  /**
+   * Set to false to disable using ssh-agent to authenticate.
+   * On Windows the only supported SSH authentication agent is Pageant.
+   */
+  readonly agent?: string;
+
+  /**
+   * The preferred identity from the ssh agent for authentication.
+   */
+  readonly agentIdentity?: string;
+
+  /**
+   * The public key from the remote host or the signing CA, used to verify the connection.
+   */
+  readonly hostKey?: string;
+
+  /**
+   * The target platform to connect to.
+   * Valid values are "windows" and "unix".
+   * If the platform is set to windows, the default script_path is c:\windows\temp\terraform_%RAND%.cmd, assuming the SSH default shell is cmd.exe.
+   * If the SSH default shell is PowerShell, set script_path to "c:/windows/temp/terraform_%RAND%.ps1"
+   * @default unix
+   */
+  readonly targetPlatform?: string;
```

Comment:
narrow down the type?
```suggestion
  readonly targetPlatform?: "unix" | "windows";
```
(`proxyScheme` below uses such a type as well)